lls/main.c

```diff
@@ -64,9 +116,24 @@ process_pkts(__attribute__((unused)) struct lls_config *lls_conf,
 			goto free_buf;
 
 		ether_type = eth_hdr->ether_type;
+		/*
+		 * Calculate offset to skip over VLAN header, if present.
+		 * TODO If VLAN stripping is enabled, we can skip this.
+		 */
+		ether_type = eth_hdr->ether_type;
+		next_hdr = (void *)((uint8_t *)(eth_hdr + 1) +
+			get_vlan_offset(eth_hdr, &ether_type));
 
-		/* TODO Add support for ARP and ND packets. */
+		/* TODO Add support for ND packets. */
 		switch (rte_be_to_cpu_16(ether_type)) {
+		case ETHER_TYPE_ARP: {
+			struct arp_hdr *arp_hdr = (struct arp_hdr *)next_hdr;
+			if (process_arp(&lls_conf->arp_cache, iface, tx_queue,
+					bufs[i], eth_hdr, arp_hdr) == -1)
+				goto free_buf;
+			/* ARP reply was sent. */
+			continue;
```

Comment:
That's OK -- it should only hit the `continue` if `process_arp()` returned 0, which indicates that the buffer was used to transmit a reply (and will be automatically freed).